src/components/common/charts/TimeRangeBarChart.tsx

```diff
@@ -223,17 +225,32 @@ const TimeRangeBarChart: React.FC<TimeRangeBarChartProps> = ({
         {
           data: xValues,
           scaleType: "band",
-          // valueFormatter: (date: Date) => date.toLocaleDateString(),
           valueFormatter: xAxisLabelFormatter,
-          tickMinStep: 3600 * 1000 * 48, // min step: 48h,
-          // If you want the label rotate
-          // tickLabelStyle: {
-          //  angle: 45,
-          //  dominantBaseline: 'hanging',
-          //  textAnchor: 'start',
-          // },
+          tickMinStep: 3600 * 1000 * 48, // min step: 48h
+          label: "Year", // x-axis label
```

Comment:
if the date range become small, the x-axis will become to "month", and if even smaller, the x-axis will be "day"